### PR TITLE
Diagnostics: Fix Position Selection

### LINKED ISSUE #4841: Particle diagnostic silently writes zero position if x y z is not explicitly specified
One of the changes in the last few months (some time between December and March) has affected the default behavior of the output for the particle data diagnostic. The new behavior seems unexpected and is not documented.

In a Full diagnostic, consider
<diag_name>.<species_name>.variables = w

Previous behavior:
Particle position is written in addition to the weight
-- this is what the documentation indicates should happen

Current behavior:
Zeros are written for the particle position regardless of where the particles actually are
The actual particle position is only written if the desired positions are explicitly specified, i.e.,
<diag_name>.<species_name>.variables = w x y z

MWE is attached

[inputs_3d.txt](https://github.com/ECP-WarpX/WarpX/files/14924231/inputs_3d.txt)

### PR DESCRIPTION
**Pull request by ax3l**
- [x] If a user specifies `<diag_name>.<species_name>.variables` without including particle position arguments we now warn.
- [x] Update missing documentation.
- [x] Close #4841: Do not write zero-positions for deselected position components. - [SetupPos](https://github.com/ECP-WarpX/WarpX/blob/development/Source/Diagnostics/WarpXOpenPMD.cpp#L711) if not needed
  - [x] Test: all positions, some positions, no positions, `none`

Follow-up to #4653

### CODE REVIEW

**Review by kale-j — APPROVED** [review]
This is now much clearer!
I tested in 2D and 3D -- seems to work as expected

**Comment by RemiLehe** [comment]
Thanks @ax3l ! Would you have time to check the `clang-tidy` error?

**Comment by ax3l** [comment]
@kale-j thank you for testing! Glad this works well! Thanks again for reporting this :sparkles: 

@RemiLehe absolutely - fixed now :)

**Comment by ax3l** [comment]
@RemiLehe when you have a second, can you approve this for the release? :)

**Comment by ax3l** [comment]
Merging, so we can do the release 24.05 :)
It is tested by @kale-j and me and only small fixes where left after Remi saw it :)

Follow-ups that we found during testing are documented in #4917

File: Source/Diagnostics/ParticleDiag/ParticleDiag.cpp [hunk 1]
```diff
@@ -28,12 +30,17 @@ ParticleDiag::ParticleDiag(
...
         // If only specific variables have been specified, fill m_plot_flags with zero and only set
         // requested variables to one
         std::fill(m_plot_flags.begin(), m_plot_flags.end(), 0);
+        bool contains_positions = false;
         if (variables[0] != "none"){
-            const std::map<std::string, int> existing_variable_names = pc->getParticleComps();
+            std::map<std::string, int> existing_variable_names = pc->getParticleComps();
+#ifdef WARPX_DIM_RZ
+            // we reconstruct to Cartesian x,y,z for RZ particle output
+            existing_variable_names["y"] = PIdx::theta;
```
Comment (ax3l) [inline Source/Diagnostics/ParticleDiag/ParticleDiag.cpp:42]: This is a hack to have `y` enabled and de-selecting was already broken before.
#4917 will do a general overhaul of all this logic and remove this assignment.

**Pull request merged**